qemu/tests/ioeventfd.py

```diff
@@ -0,0 +1,164 @@
+"""
+ioeventfd.py include following case:
+    1. Test ioeventfd under stress.
+    2. Check the ioeventfd property.
+"""
+import logging
+import re
+
+from virttest import error_context
+from virttest import env_process
+from virttest import utils_test
+from virttest import qemu_qtree
+from virttest import utils_misc
+from avocado.utils import process
+from provider.storage_benchmark import generate_instance
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test the related functions of ioeventfd.
+    Step:
+        Scenario 1:
+            1.1 Start guest with ioeventfd=off.
+                For windows: Check whether vioscsi.sys verifier enabled in
+                guest.
+            1.2 Dd file in the guest.
+                For rhel: dd if=/dev/zero of=/home/test bs=1M count=4096
+                oflag=direct.
+                For windows: We can load some big data to the disk via
+                DummyCMD.exe tools or other tools.
+            1.3 Reboot guest.
+            1.4 Iozone(Iozone.exe) -a in the guest.
+            1.5 Poweroff guest.
+            1.6 Repeat step 1.1-1.5 with ioeventfd=on.
+        Scenario 2:
+            2.1 Boot guest with ioeventfd=off.
+            2.2 Execute info qtree in QMP monitor, info qtree should show the
+            ioeventfd = false.
+            2.3 Check the ioeventfd=off via /proc/$PID/fd/.
+            2.4 Boot guest with ioeventfd=on.
+            2.5 Execute info qtree in QMP monitor, info qtree should show the
+            ioeventfd = true.
+            2.6 Check the ioeventfd=on via /proc/$PID/fd/.
+            2.7 Compare the output of 'ls -l /proc/$PID/fd/', the fds with
+            "off" should be less than the one with "on".
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _common_io_test(session):
+        """
+        Execute common io testing inside guest.
+        """
+        timeout = float(params['op_timeout'])
+        op_cmd = params['op_cmd']
+        if params['os_type'] == 'windows':
+            op_cmd = utils_misc.set_winutils_letter(session, op_cmd)
+        o = session.cmd(op_cmd, timeout)
+        logging.debug(o)
```

Comment:
can we add a "status" to judge whether it is succesfull?

Reply:
this method session.cmd will judge whether it is successful.